src/handlers/movies/getMovies.ts

```diff
@@ -0,0 +1,97 @@
+import sortBy from 'lodash/sortBy';
+import getAllMovies from '../../models/Movie/getAll';
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import getAllGenres from '../../models/Genre/getAll';
+import { MovieModel } from '../../models/Movie/type';
+import { DEFAULT_CONTENTFUL_LIMIT } from '../../utils/contentful';
+import { CONTENTFUL_INCLUDE } from '../../types/contentful';
+import { serverErrorResponse } from '../../utils/api/apiResponses';
+
+export const handler: APIGatewayProxyHandler = async (event) => {
+	try {
+		const searchFilters: {
+			page?: number;
+			limit?: number;
+			genre?: string;
+			search?: string;
+		} = {};
+
+		const { page, limit, genre, search } = event?.queryStringParameters || {};
+
+		if (page) {
+			const parsedPage = parseInt(page, 10);
+			if (!isNaN(parsedPage) && parsedPage > 0) {
+				searchFilters.page = parsedPage;
+			}
+		}
+
+		if (limit) {
+			const parsedLimit = parseInt(limit, 10);
+			if (!isNaN(parsedLimit) && parsedLimit > 0) {
+				searchFilters.limit = parsedLimit;
+			}
+		}
+
+		if (genre) {
+			searchFilters.genre = genre;
+		}
+
+		if (search) {
+			searchFilters.search = search;
+		}
+
+		if (searchFilters.genre) {
+			// 	we need to use the other getAll function because we need to filter by genre and Contentful doesn't support filtering by reference fields many to many
+			const moviesWithGenreEntries = await getAllGenres({
+				include: CONTENTFUL_INCLUDE.moviesWithGenre,
+				query: {
+					title: searchFilters.genre,
+				},
+			});
```

Comment:
There's a possibility of this api call to return an empty array in `data`
ie: `GET localhost:3000/dev/movies?genre=actionzzz`

This will need a check for this scenario, otherwise it's throwing a 500 error.

Reply:
Nice! Good catch here